sertmer/GlobeTrotter
# Expo react native maps

### What's this PR do?
This PR adds the [react-native-maps](https://github.com/react-native-community/react-native-maps) library.  Currently the app only renders [MapView](https://github.com/react-native-community/react-native-maps/blob/master/docs/mapview.md) and a [Marker](https://github.com/react-native-community/react-native-maps/blob/master/docs/marker.md).  The Maps component also renders an Agenda component provided by [react-native-calendars](https://github.com/wix/react-native-calendars)
### Where should the reviewer start?
The GlobeTrotter frontend uses data that is provided by the [globe_trotter_api](https://github.com/johnktravers/globe_trotter_api).  All the data used by the Maps component is fetched in the Trips component, and is passed to Maps using the navigation method provided by the react-native-navigation library.  From the Tips component the data is used in TripPreview before being passed to the Maps component when a destination is pressed.
### How should this be manually tested?
**There may be issues running the application in a web browser.  Please visit the App Store or Play Store and download the Expo app by Expo Project to run this application on a mobile device.**

In order to manually test the Maps component:

- pull down the ```expo-react-native-maps``` branch

- run ```npm i``` then ```npm start```

- Once the expo development server opens in the web browser and you see ```tunnel ready``` scan the QR code using the Expo app QR scanner.  This should begin a javascript bundle process and the app should load on your mobile device.

- Once the application loads and you see a list of trips, select any trip to be taken to a trip preview that will list all the destinations for that specific trip.  Click a destination to be taken to the Map component.

- The Maps component should display a map and marker for that specific destination, along with an Agenda component, the Agenda component is blank and a loader will spin continuously as I have not defined styles for a day.   

### Any background context you want to provide?
There was some (some = A LOT!!) confusion on my part when adding the react-native-maps library.  According to the [docs](https://docs.expo.io/versions/latest/sdk/map-view/) if you wish to use this dependency you **must** have a Google Cloud Platform account and **follow all** the configuration steps in order to make this work in your application.  However if your backend team is amazing, which ours is, and they provide you with all the data you need you can skip all configuration steps and just supply the components the data they need to render correctly.  

As a side note this was the third react-native-maps library we installed it is very important to research the library and framework you are using.  React-native is in a constant state of flux as it grows meaning there are a lot of old unmaintained libraries out there.  Checking the latest commit can quickly tell you if the library has been kept up to date.
### What are the relevant tickets?
#29 
### Screenshots:
This first screen shot shows a partially rendered day in the agenda component.  The second is what happens when no style or data is provided.
![Screenshot_20200223-200939](https://user-images.githubusercontent.com/45470456/75126798-d05c5e00-5678-11ea-9df5-e4bec04ad858.png)
![Screenshot_20200223-200927](https://user-images.githubusercontent.com/45470456/75126806-d9e5c600-5678-11ea-9337-7a163c02cdfa.png)
